Sources/Regex/Atom.swift

```diff
@@ -450,4 +450,22 @@ extension Atom {
 
     }
   }
+
+  /// Retrieve the character value of the atom if it represents a literal
+  /// character or unicode scalar, nil otherwise.
+  var literalCharacterValue: Character? {
+    switch self {
+    case .char(let c):
+      return c
+    case .scalar(let s):
+      return Character(s)
```

Comment:
I'm a little worried this won't work in grapheme semantic mode. We want to permit a sequence of literal scalars that identify a single grapheme cluster.

Reply:
Ah good point

Reply:
Later, this is a semantic concern so it's weird for it to have any presence in the AST or parser. An AST can produce a model type given a set of parameters. It's possible those parameters come in from the literal itself as options, in which case the parser stores set of options, or it's parameters to the function that produces the model type. The latter is the more general formulation anyways, so I'd go that way.